multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/a3mangatheme/A3MangaTheme.kt

```diff
@@ -0,0 +1,209 @@
+package eu.kanade.tachiyomi.multisrc.a3mangatheme
+
+import android.util.Base64
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.POST
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
+import eu.kanade.tachiyomi.source.model.Page
+import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.ParsedHttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.FormBody
+import okhttp3.Headers
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import org.jsoup.nodes.Element
+import rx.Observable
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+import javax.crypto.Cipher
+import javax.crypto.SecretKeyFactory
+import javax.crypto.spec.IvParameterSpec
+import javax.crypto.spec.PBEKeySpec
+import javax.crypto.spec.SecretKeySpec
+
+open class A3MangaTheme(
+    override val name: String,
+    override val baseUrl: String,
+    override val lang: String,
+) : ParsedHttpSource() {
+
+    override val supportsLatest: Boolean = false
+
+    override val client: OkHttpClient = network.cloudflareClient
+
+    override fun headersBuilder(): Headers.Builder = Headers.Builder().add("Referer", "$baseUrl/")
+
+    override fun popularMangaRequest(page: Int): Request = GET("$baseUrl/page/$page/", headers)
+
+    override fun popularMangaSelector() = ".comic-list .comic-item"
+
+    override fun popularMangaFromElement(element: Element) = SManga.create().apply {
+        setUrlWithoutDomain(element.select(".comic-title-link a").attr("href"))
+        title = element.select(".comic-title").text().trim()
+        thumbnail_url = element.select(".img-thumbnail").attr("abs:src")
+    }
+
+    override fun popularMangaNextPageSelector() = "li.next:not(.disabled)"
+
+    override fun latestUpdatesRequest(page: Int) = throw Exception("Not used")
```

Comment:
Here and in other places. Could also put all the unused stuff at the same place.
```suggestion
    override fun latestUpdatesRequest(page: Int) = throw UnsupportedOperationException("Not used")
```